[FFDW][UXIT-2419] Setup Digest structure to allow for multiple Digest issues [skip percy]

barbaraperic opened the pull request:
## 📝 Description

This PR restructures the digest section to organize content by issues, introducing a hierarchical digest issue → articles architecture.

Digest Information Architecture
•  Restructured routing: `/digest → /digest/[issue] → /digest/[issue]/[slug]`

Content Migration
•  Moved all digest articles to `src/content/digest/articles/*.md`
•  Added first digest issue file at `src/content/digest/issues/1.md`

New Components & Utils
•  Added new path constants for `articlesPath`, `issuePath`, `issueUrl()`, and `articleUrl()`
•  Created `DigestIssueFrontmatterSchema` for issue metadata validation
•  Added `getDigestIssueData() `and `getDigestArticlesWithIssueContext()` utilities


**NOTE**: Since this PR is already quite big, I plan to make updates to ff-site in another PR.

- **Type:** New feature / Refactor

## 🛠️ Key Changes

Previously, digest articles had their own `issue-number` and `article-number` fields in their frontmatter, leading to potential inconsistencies and duplications. This created a maintenance burden where article metadata could become out of sync with the actual digest issue structure.

This PR introduces centralized article organization by moving article metadata management to the digest `issue` level. Articles are now organized as a list within each digest issue, with the issue number and article number being derived from the issue context rather than stored in individual article files.

- Article Creation: Focus on content, authors, and metadata
- Issue Organization: Focus on curating and ordering articles for publication

### Example Workflow
1. Create Articles: Write 5 articles about decentralization topics
2. Create Issue: Create "Issue 3: The Future of Decentralization"
3. Organize Articles:
  - Add "Article 1: Blockchain Basics" (article-number: 1)
  - Add "Article 2: Privacy in Web3" (article-number: 2)
  - Add "Article 3: Decentralized Governance" (article-number: 3)
4. Publish: The issue automatically displays articles in the specified order

## 📌 To-Do Before Merging

- [x] Fix Cypress
- [ ]  Handle deleted articles
  - Create a cleanup script that removes non-existent articles from issue files.
  - Can be a follow-up PR — for now, the current logic simply skips missing articles and doesn’t display them.

## 🧪 How to Test

  1. Go to decapCMS
  2. Update issues or create a new one (pretend to be an editor - does the flow make sense, does it need more hints)
  3. Check if correctly reflects the UI

## 📸 Screenshots

### CMS Issue with added articles

<img width="1259" height="761" alt="Screenshot 2025-10-22 at 09 30 58" src="https://github.com/user-attachments/assets/885c116c-4ad0-4437-950d-3e32ddafd702" />

mirhamasala commented on apps/ffdweb-site/public/admin/config.yml:
```diff
@@ -146,10 +146,63 @@ collections:
 
+  - name: "digest_issues"
+    label: "Digest Issues"
+    folder: "apps/ffdweb-site/src/content/digest-issues/"
```
Maybe?

```suggestion
    folder: "apps/ffdweb-site/src/content/digest/issues/"
```

mirhamasala commented on apps/ffdweb-site/public/admin/config.yml:
```diff
@@ -146,10 +146,63 @@ collections:
+    fields:
+      - name: "issue-number"
+        label: "Issue Number"
+        widget: "number"
+        min: 1
```
This can be constrained to just ["1", "2"] I believe..  https://decapcms.org/docs/widgets/#select

mirhamasala commented on apps/ffdweb-site/src/app/digest/schemas/DigestIssueFrontmatterSchema.ts:
```diff
@@ -0,0 +1,27 @@
+// apps/ffdweb-site/src/app/digest/schemas/DigestIssueFrontmatterSchema.ts
```
```suggestion
```

CharlyMartin commented on apps/ffdweb-site/src/app/_constants/paths.ts:
```diff
@@ -26,17 +26,26 @@ export type PathValues = StaticPath | DynamicPathValues
+    ...createPathConfig('/digest', 'Digest'),
+    articlesPath: `${CONTENT_ROOT}/digest/articles`,
+    issuePath: `${CONTENT_ROOT}/digest/issues`,
+    issueUrl: (issueNumber: string) => `/digest/issue-${issueNumber}` as const,
```
And same for `articleUrl`

CharlyMartin commented on apps/ffdweb-site/src/content/digest/creating-human-records-that-stand-the-test-of-time.md:
```diff
@@ -3,8 +3,6 @@ title: Creating Human Records that Stand the Test of Time
 created-on: 2023-10-16T17:14:41.944Z
 updated-on: 2023-10-16T17:14:41.951Z
 published-on: 2023-10-16T17:14:41.958Z
-issue-number: "1"
```
Could `issue-number` also be a number?

```suggestion
issue-number: 1
```

CharlyMartin commented on apps/ffdweb-site/src/app/digest/utils/getDigestArticlesWithIssueContext.ts:
```diff
@@ -0,0 +1,59 @@
+  ).then((results) => results.filter(Boolean))
+
+  return articles
+}
```
I don't know if it works, just a suggestion:


```suggestion
export async function getAllDigestArticlesWithIssueContext({
  issueNumber,
}: getAllDigestArticlesWithIssueContextProps) {
  try {
    const digestIssue = await getDigestIssueData(issueNumber)

    const articles = await Promise.all(
      digestIssue.articles.map(async (articleData, index) => {
        const article = await getDigestArticleData(articleData.title)
        return {
          ...article,
          issueNumber: digestIssue.issueNumber,
          articleNumber: articleData.articleNumber ?? index + 1,
        }
      }),
    )

    return articles
  } catch {
    return []
  }
}
```

CharlyMartin replied:
My understanding is that Zod throws an error when validation fails, it's up to us to catch/handle it.

How about we block 30 minutes tomorrow to pair and close this?

CharlyMartin reviewed (commented):
That's a huge lift Barbara! Well done!

I left some comments here and there. Maybe we can pair to go through them?

barbaraperic commented:
Hey @CharlyMartin based on our discussion this morning, I have:
- added metadata to `digest/[issue]`
- fixed the layout on `digest` page, so that issue 2 comes before issue 1
- added cypress test for `digest_issue`
- also extracted parsing params in its own file `parseDigestParams.ts`
